app/src/main/java/com/hifeful/mealmania/presentation/myMeals/MyMealsViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.hifeful.mealmania.presentation.myMeals
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hifeful.mealmania.domain.repository.MealsRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class MyMealsViewModel @Inject constructor(
+    private val mealsRepository: MealsRepository
+) : ViewModel() {
+
+    val events = Channel<MyMealsEvent>(Channel.UNLIMITED)
+    val state = MutableStateFlow(MyMealsState())
+
+    init {
+        handleEvents()
+    }
+
+    private fun handleEvents() {
+        viewModelScope.launch {
+            events.consumeAsFlow()
+                .collect { event ->
```

Comment:
Why you don't use `onEach` instead?

Reply:
Because it won't work like that. I can reimplement using `onEach` and `launchIn` operator:


```kotlin
  events.consumeAsFlow()
      .onEach { event ->
          when (event) {
            // ...
      }
      .launchIn(viewModelScope)
```